lib/internal/streams/destroy.js

```diff
@@ -52,6 +54,11 @@ function destroy(err, cb) {
     r.destroyed = true;
   }
 
+  if (cb) {
```

Comment:
Perhaps here and elsewhere in this function we should instead check `typeof cb === 'function'`?

Reply:
We don't do that anywhere else in this file. If you'd prefer that I would prefer to handle that in a separate PR and change it everywhere in this file.